### PR TITLE
plugin/k8s: fix pods disabled behavior

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. What does this pull request do?

Fix return code for pods queries when pods are disabled - per docs, should be `NXDOMAIN`, not `SERVFAIL`...  Unit tests also expect `NXDOMAIN` (but the return code not actually checked in the tests due the the "expected" error returned)

### 3. Which documentation changes (if any) need to be made?

### CODE REVIEW

**Comment by chrisohaver** [comment]
I didn't get the integration tests right... missing expected authority section...

**Comment by chrisohaver** [comment]
Added expected auth section in coredns/ci#4 ... retest integration when merged.

File: plugin/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func (k *Kubernetes) Lookup(state request.Request, name string, typ uint16) (*dn
 
 // IsNameError implements the ServiceBackend interface.
 func (k *Kubernetes) IsNameError(err error) bool {
-	return err == errNoItems || err == errNsNotExposed || err == errInvalidRequest
+	return err == errNoItems || err == errNsNotExposed || err == errInvalidRequest || err == errPodsDisabled
```
Comment (miekg) [inline plugin/kubernetes/kubernetes.go]: This smells like a hack. errPodsDisabled should be used in a general IsNameError function.

Comment (chrisohaver) [reply plugin/kubernetes/kubernetes.go]: not following... (i.e. why do you think its a hack?)

Comment (miekg) [reply plugin/kubernetes/kubernetes.go]: What has podDisabled to do with nxdomain? errInvalidRequest is already border line

Comment (chrisohaver) [reply plugin/kubernetes/kubernetes.go]: If the podMode is set to podsDisabled (default setting in k8s plugin), then we don't produce `A` records for pods (e.g. 1-2-3-4.ns.pod.cluster.local)...

If some one tries to query a pod in this mode, then there will be no `A` record for it. `NXDOMAIN` seems like the correct answer here.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] plugin/k8s: f..." ]
>chrisohaver commented on this pull request.
>
>
>
>> @@ -143,7 +143,7 @@ func (k *Kubernetes) Lookup(state request.Request, name string, typ uint16) (*dn
>
> // IsNameError implements the ServiceBackend interface.
> func (k *Kubernetes) IsNameError(err error) bool {
>-	return err == errNoItems || err == errNsNotExposed || err == errInvalidRequest
>+	return err == errNoItems || err == errNsNotExposed || err == errInvalidRequest || err == errPodsDisabled
>
>If the podMode is set to podsDisabled (default setting in k8s plugin), there we don't produce A records for pods (e.g. 1-2-3-4.ns.pod.cluster.local)...
>
>If some one tries to query a pod in this mode, then there will be no `A` record for it. `NXDOMAIN` seems like the correct answer here.

There will be no records, so return errNoItems?

**Comment by chrisohaver** [comment]
> return errNoItems?

OK - that works too...

**Pull request merged**